pkgs/development/python-modules/flask-basicauth/default.nix

```
@@ -0,0 +1,42 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, fetchpatch
+, flask
+, python
+}:
+
+buildPythonPackage rec {
+  pname = "flask-basicauth";
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "jpvanhal";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-han0OjMI1XmuWKHGVpk+xZB+/+cpV1I+659zOG3hcPY=";
+  };
+
+  patches = [
+    (fetchpatch {
+      name = "fix-test-flask-ext-imports.patch";
+      url = "https://github.com/jpvanhal/flask-basicauth/commit/23f57dc1c3d85ea6fc7f468e8d8c6f19348a0a81.patch";
+      sha256 = "sha256-njUYjO0TRe3vr5D0XjIfCNcsFlShbGxtFV/DJerAKDE=";
+    })
+  ];
+
+  propagatedBuildInputs = [ flask ];
+
+  checkPhase = ''
+    ${python.interpreter} -m unittest discover
+  '';
```

When overriding a phase, you should always run the pre and post hooks for that given phase. I know a lot of Python packages don't do this, but that will eventually be fixed.

```suggestion
  checkPhase = ''
    runHook preCheck
    ${python.interpreter} -m unittest discover
    runHook postCheck
  '';
```

---

thank you for catching this. i'm still learning!